src/renderer.js

```diff
@@ -45,7 +45,12 @@ export class Renderer {
 
   static buildContext(elementId, backend, width, height, background) {
     const renderer = new Renderer(elementId, backend);
-    if (width && height) { renderer.resize(width, height); }
+    if (width && height) {
+      if (backend === Renderer.Backends.CANVAS) {
+        [width, height] = CanvasContext.CheckCanvasDimensionsForBrowserLimit(width, height);
```

Comment:
Looks like you're already calling this inside `resize`, so we don't need this here, no?

Reply:
ah, right, i thought this was separate. removing this call.